relay-server/src/processing/logs/process.rs

```diff
@@ -17,6 +20,23 @@ use crate::services::outcome::DiscardReason;
 pub fn expand(logs: Managed<SerializedLogs>, _ctx: Context<'_>) -> Managed<ExpandedLogs> {
     let trust = logs.headers.meta().request_trust();
 
+    #[cfg(feature = "processing")]
+    let retentions = &ctx
+        .project_info
+        .config
+        .retentions
+        .as_ref()
+        .and_then(|r| r.retention_for_trace_item(TraceItemType::Log));
+    #[cfg(feature = "processing")]
+    let retention = retentions.map_or(_ctx.project_info.config.event_retention, |r| {
+        Some(r.standard)
+    });
+    #[cfg(feature = "processing")]
+    let downsampled_retention = retentions
+        .map_or(_ctx.project_info.config.downsampled_event_retention, |r| {
+            Some(r.downsampled)
+        });
```

Comment:
This would benefit from a helper function on `ProjectConfig` that returns the fallback, something like

```rust
impl ProjectConfig {
    pub fn get_retentions(&self, ty: TraceItemType) -> (u16, u16) {
         match self.get_specific_retentions(ty) {
             Some(r) => (r.standard, r.downsampled)
             None => (self.event_retention, self.downsampled_event_retention)
        }
    }

    fn get_specific_retentions(&self, ty: TraceItemType) -> Option<&RetentionSettings> {
         self.retentions?.retention_for_trace_item(ty)
    }
}
```

Reply:
Indeed it would, but taking ownership of `retentions` and trying to extract the retentions proved to be too much for me.